### PR TITLE
Generic helm chart

### LINKED ISSUE #242: Generic helm chart
Implement a general purpose helm chart for zilla, such that it can be reused by our zilla-examples, but without any special knowledge of how the zilla-examples work.

Verify the helm chart works for zilla-examples.

Automatically publish to ArtifactHub in reaction to a new zilla git flow release to main branch, via GitHub Actions.

Update zilla-examples to use deployed helm chart from ArtifactHub.

### PR DESCRIPTION
**Pull request by attilakreiner**
## Description

This change implements a general purpose helm chart for zilla, such that it can be reused by our zilla-examples, but without any special knowledge of how the zilla-examples work.

Fixes #242

### CODE REVIEW

File: cloud/helm-chart/pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <licenses>
+        <license>
+            <name>Aklivity Community License Agreement</name>
+            <url>https://www.aklivity.io/aklivity-community-license/</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+
+    <build>
+        <plugins>
```
Comment (jfallows) [inline cloud/helm-chart/pom.xml:29]: Please add the license check goal.

File: cloud/helm-chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: zilla
+description: Zilla The Event-Driven API Gateway
+type: application
+version: 0.1.0
+appVersion: "0.9.42"
```
Comment (jfallows) [inline cloud/helm-chart/Chart.yaml:6]: This is hard coded, but should come from build version so it is always up-to-date as part of the release.

File: cloud/helm-chart/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Install
+
+```
+$ helm install zilla . --namespace zilla --create-namespace --wait #--values your.custom.values.yaml
```
Comment (jfallows) [inline cloud/helm-chart/README.md:4]: Should this example show how to specify file for zilla.yaml too?
(same for upgrade)

File: cloud/helm-chart/values.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+securityContext: {}
+
+livenessProbePort: 0
+readinessProbePort: 0
+
+args:
+  - start
+  - -v
+  - -e
```
Comment (jfallows) [inline cloud/helm-chart/values.yaml:24]: Do you think it makes sense to include the `-v` (verbose) for a production helm chart?

Comment (attilakreiner) [reply cloud/helm-chart/values.yaml:24]: I don't think it's bad to see what's going on in a prod system. Unless there is some concrete disadvantage (e.g. performance issue), I'd keep it.

File: cloud/helm-chart/README.md (hunk 3 above)
Comment (attilakreiner) [reply cloud/helm-chart/README.md]: fixed

File: cloud/helm-chart/pom.xml (hunk 1 above)
Comment (attilakreiner) [reply cloud/helm-chart/pom.xml:29]: updated, pls chk

File: cloud/helm-chart/Chart.yaml (hunk 2 above)
Comment (attilakreiner) [reply cloud/helm-chart/Chart.yaml]: updated, pls chk

**Review by jfallows — APPROVED** [review]

**Pull request merged**